Better arch detection, based on the original libdecaf, from which this
library is deriverd.

https://sourceforge.net/p/ed448goldilocks/code/ci/master/tree/src/CMakeLists.txt#l49

Also remove useless variables.

--- configure.ac.orig	2020-03-16 08:25:44.000000000 +0000
+++ configure.ac
@@ -45,40 +45,41 @@ AC_CANONICAL_HOST
 # target arch dirs:
 # availables: arch_32, arch_arm_32, arch_neon, arch_ref64, arch_x86_64
 AS_CASE([$host_cpu],
-  [ia64|mips64|mips64eb|mipseb64|mips64el|mipsel64|mips64*|powerpc64*|sparc64|x86_64*|amd64*], [ARCH_DIR=arch_x86_64],
-  [arch64|aarch64|powerpc*], [ARCH_DIR=arch_ref64],
-  [arm64], [ARCH_DIR=arch_neon],
-  [arm32|armv*], [ARCH_DIR=arch_arm_32],
+  [i?86], [ARCH_DIR=arch_32],
+  [x86_64|amd64], [ARCH_DIR=arch_x86_64],
+  [armv6], [ARCH_DIR=arch_arm_32],
+  [armv7], [ARCH_DIR=arch_neon],
+  [aarch64], [ARCH_DIR=arch_ref64],
   [ARCH_DIR=arch_32]
 )
 
+# This is used to enable arch specific flags for x86 CPUS.
+AS_CASE([$host_cpu],
+  [i?86], [ARCH_TYPE=X86],
+  [x86_64|amd64], [ARCH_TYPE=X86],
+  [ARCH_TYPE=non-X86]
+)
+
 AC_SUBST([ARCH_DIR])
-AC_SUBST([ARCH_ARM])
+AC_SUBST([ARCH_TYPE])
 
 AS_IF([test "x$ARCH_DIR" = "xarch_x86_64"], [needx64=yes],
     [test "x$ARCH_DIR" != "xarch_x86_64"], [needx64=no])
 
-AM_CONDITIONAL([X86], [test "x$needx64" = "xyes"])
-
 AS_IF([test "x$ARCH_DIR" = "xarch_ref64"], [need64=yes],
     [test "x$ARCH_DIR" != "xarch_ref64"], [need64=no])
 
-AM_CONDITIONAL([ARCH_64], [test "x$need64" = "xyes"])
-
 AS_IF([test "x$ARCH_DIR" = "xarch_neon"], [needneon=yes],
     [test "x$ARCH_DIR" != "xarch_neon"], [needneon=no])
 
-AM_CONDITIONAL([ARCH_NEON], [test "x$needneon" = "xyes"])
-
 AS_IF([test "x$ARCH_DIR" = "xarch_arm_32"], [needarm32=yes],
     [test "x$ARCH_DIR" != "xarch_arm_32"], [needarm32=no])
 
-AM_CONDITIONAL([ARCH_ARM_32], [test "x$needarm32" = "xyes"])
-
 AS_IF([test "x$ARCH_DIR" = "xarch_32"], [need32=yes],
     [test "x$ARCH_DIR" != "xarch_32"], [need32=no])
 
-AM_CONDITIONAL([ARCH_32], [test "x$need32" = "xyes"])
+AM_CONDITIONAL([ARCH_X86], [test "x$ARCH_TYPE" = "xX86"])
+AM_CONDITIONAL([ARCH_NEON], [test "x$ARCH_DIR" = "xarch_neon"])
 
 AX_CFLAGS_GCC_OPTION([-Wall])
 AX_CFLAGS_GCC_OPTION([-Wextra])
